libraries/AP_Logger/AP_Logger_Backend.cpp

```diff
@@ -593,8 +593,8 @@ uint16_t AP_Logger_Backend::log_num_from_list_entry(const uint16_t list_entry)
     }
 
     uint32_t log_num = oldest_log + list_entry - 1;
-    if (log_num > MAX_LOG_FILES) {
-        log_num -= MAX_LOG_FILES;
+    if (log_num > (uint32_t)_front._params.max_log_files.get()) {
```

Comment:
we need the casting there to prevent a compiler error about comparison between signed and unsigned. max_log_files is int16